scripts/g.extension/testsuite/test_addons_download.py

```diff
@@ -0,0 +1,164 @@
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        # Modules with non-standard branch would be good for testing...
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        for file in self.files:
+            self.assertFileExists(file)
+
+    def test_gitlab_install(self):
+        """Test installing extension from gitlab"""
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://gitlab.com/vpetras/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        for file in self.files:
+            self.assertFileExists(file)
+
+    def test_bitbucket_install(self):
+        """Test installing extension from bitbucket"""
+        d_rast_files = [
+            os.path.join(self.install_prefix, "scripts", "r.sim.stats"),
+            os.path.join(self.install_prefix, "docs", "html", "r.sim.stats.html"),
+        ]
+        self.assertModule(
+            "g.extension",
+            extension="r.sim.stats",
+            url="https://bitbucket.org/lrntct/r.sim.stats",
+            prefix=self.install_prefix,
+        )
+
+        for file in d_rast_files:
+            self.assertFileExists(file)
+
+    def test_github_install_official(self):
+        """Test installing C-extension from official addons repository"""
+        r_gdd_files = [
+            os.path.join(self.install_prefix, "bin", "r.gdd"),
+            os.path.join(self.install_prefix, "docs", "html", "r.gdd.html"),
+        ]
+        self.assertModule("g.extension", extension="r.gdd", prefix=self.install_prefix)
+
+        for file in r_gdd_files:
+            self.assertFileExists(file)
+
+    def test_github_install_official_multimodule(self):
+        """Test installing multi-module extension from official addons repository"""
+        i_sentinel_files = [
+            os.path.join(
+                self.install_prefix, "scripts", "i.sentinel.parallel.download"
+            ),
+            os.path.join(
+                self.install_prefix, "docs", "html", "i.sentinel.parallel.download.html"
+            ),
+            os.path.join(self.install_prefix, "scripts", "i.sentinel.import"),
+            os.path.join(self.install_prefix, "docs", "html", "i.sentinel.import.html"),
+        ]
+        self.assertModule(
+            "g.extension", extension="i.sentinel", prefix=self.install_prefix
+        )
+
+        for file in i_sentinel_files:
+            self.assertFileExists(file)
+            if not file.endswith("html"):
+                print(file)
+                self.assertModule(file, **{"help": True})
+
+    def test_windows_install(self):
+        """Test function for installing extension on MS Windows"""
```

Comment:
This function really needs more info/motivation why it is here. You can put it in the docstring.